File: 02-spring-boot-data/spring-data-core/src/main/java/vip/yazilim/basics/springdatacore/service/IPersonService.java
package vip.yazilim.basics.springdatacore.service;

import java.util.List;

import vip.yazilim.basics.springdatacore.entity.Person;

public interface IPersonService {

    // ...
    Person insert(Person person);

    Person update(Person person);

    void delete(int id);

    // ...
    List<Person> findAll();

    Person findById(int id);

    Person findFirstByNameAsc();
    
    List<Person> findByName(String name);
    
    List<Person> findByLocation(String location);

}

File: 03-spring-boot-thymeleaf/04-thymeleaf-crud/src/main/kotlin/vip/yazilim/thymeleaf/crud/service/IPersonService.kt
package vip.yazilim.thymeleaf.crud.service

import vip.yazilim.thymeleaf.crud.model.Person

/**
 *
 * @author maemresen - maemresen@yazilim.vip
 * 27.12.2020
 */
interface IPersonService {
    fun getAll(): List<Person>
    fun getByName(name: String): Person
}